extra_geom/base.py

```diff
@@ -794,6 +794,77 @@ def data_coords_to_positions(self, module_no, slow_scan, fast_scan):
             + (np.expand_dims(tile_ss, -1) * coords_ss_vec) \
             + (np.expand_dims(tile_fs, -1) * coords_fs_vec)
 
+    def positions_to_data_coords(self, positions,  origin=None, unit=None):
+        """Convert assembled positions to data array coordinates
+
+        Data array coordinates are how you might refer to a pixel in an array of
+        detector data: module number, and indices in the slow-scan and fast-scan
+        directions. But coordinates in the two pixel dimensions aren't necessarily
+        integers, e.g. if they refer to the centre of a peak.
+
+        Positions is a 2D array (n, 2) describing (x, y) positions of points within an
+        assembled detector image. The may hold floats for sub-pixel positions.
+
+        If origin is None (default) the position indices are relative to the centre of
+        the geometry. `origin=(0, 0)` would set the origin to the first index of the
+        assembled array.
+        If unit is None (default) the positions given are in pixel, else given positions
+        are in meter relative to the centre, e.g. `unit=1e-3` interprets positions in mm.
+
+        .. note::
+
+          This uses an assembly limited to snapping pixel to a regular grid, so any
+          subpixel offset of tile or rotation in your geometry will be ignored.
```

Comment:
I think it should be possible - and not much more complex - to do this 'properly', preserving subpixel offsets and rotations and so on. Something like (pseudocode):

```python
for mod in modules:
    for t, tile in enumerate(mod):
        ss_coord, fs_coord = find_data_coords_for_tile(data, tile)
        on_tile = (0 < ss_coord) & (ss_coord < tile.ss_pixels) & (0 < fs_coord) & (fs_coord < tile.fs_pixels)
        ss_coord, fs_coord = tile_coords_to_module(ss_coord[on_tile], fs_coord[on_tile], t)
        # Record these results to result arrays
```

I'm assuming the numbers of tiles per detector are small enough to get away with just checking each tile one by one. So far it's a max of 128 tiles (LPD-1M), IIRC. If that's too slow, we'd need a bit more complexity to find the relevant tiles efficiently, but hopefully not much more.